kubernetes/data_source_kubernetes_pod.go

```diff
@@ -0,0 +1,72 @@
+package kubernetes
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log"
+)
+
+func dataKubernetesPod() *schema.Resource {
+	podSpecFields := podSpecFields(false, false, false)
+	// Setting this default to false prevents a perpetual diff caused by volume_mounts
+	// being mutated on the server side as Kubernetes automatically adds a mount
+	// for the service account token
+	return &schema.Resource{
+		Read: dataKubernetesPodRead,
+
+		Schema: map[string]*schema.Schema{
+			"metadata": namespacedMetadataSchema("pod", true),
+			"spec": {
+				Type:        schema.TypeList,
+				Description: "Specification of the desired behavior of the pod.",
+				Computed:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: podSpecFields,
+				},
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataKubernetesPodRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*KubeClientsets).MainClientset
+
+	name := d.Get("metadata.0.name").(string)
```

Comment:
We're trying to move away from the format `metadata.0.name`, since this syntax is deprecated in the terraform-plugin-sdk. I just learned about this today, actually. It turns out there is a new syntax you can use. See this line for an example: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/master/kubernetes/resource_kubernetes_config_map.go#L48

I know the old syntax is still found in many places in this provider, so it makes sense that you used it too. We'll have to refactor those old ones at some point.